code/datums/components/riding/riding_mob.dm

```diff
@@ -216,6 +218,15 @@
 	else if(ride_check_flags & CARRIER_NEEDS_ARM) // fireman
 		human_parent.buckle_lying = 90
 
+/datum/component/riding/creature/post_vehicle_mob_buckle(mob/living/ridden, mob/living/rider)
+	if(!require_minigame|| ridden.faction.Find(REF(rider)))
+		return
+	ridden.Shake(duration = 2 SECONDS)
+	ridden.balloon_alert(rider, "tries to shake you off.")
+	var/datum/riding_minigame/game = new
+	INVOKE_ASYNC(game, TYPE_PROC_REF(/datum/riding_minigame, commence_minigame), ridden, rider)
```

Comment:
Might be mildly cleaner if you passed in `ridden` and `rider` in `New` rather than needing to pass them in `commence`? 

Likewise `commence_minigame` could be `set waitfor = FALSE` so you don't need to worry about asyncing it?